resources/pre-upgrade.sh

```diff
@@ -33,17 +33,36 @@ function dumpData(){
       local USERS
       USERS="$(mysql -uroot mysql -e "select GROUP_CONCAT(User) FROM user WHERE NOT User LIKE '%mysql.%' AND NOT User='root';" | tail -n +2 | sed 's/,/ /g')"
 
-      local user
-      for user in ${USERS}
-      do
-          local CREATE
-          CREATE="$(mysql --skip-column-names -A mysql -e "SET @@SESSION.print_identified_with_as_hex = 1; SHOW CREATE USER '${user}'")"
-          echo "${CREATE};" >> /alldb.sql
-      done
+      # as users may exists within the dump but need to be recreated with all privileges the user must be dropped first
```

Comment:
Thank you. The comment makes it somewhat clearer but I still would encourage you to extract it into its own function.